datalad/distribution/create_publication_target_sshwebserver.py

```diff
@@ -0,0 +1,270 @@
+                # - what about just adding new push url to that remote?
+                # - if url(s) fit, then still try to create target?
+
+                # cmd = ["git", "remote", "get-url", "--push", remote]
+                # out, err = runner.run(cmd, cwd=handle_repo.path)
+                # if handle_remote_url_push != out.strip():
+                lgr.warning("Sibling {0} already exists. Skipping".format(target))
+                continue
+
+            # %NAME
+            REPO_NAME = repo.replace("/", "-")
+
+            # create remote repository
+            path = target_dir.replace("%NAME", REPO_NAME)
+
+            if path != '.':
+                # check if target exists, and if not --force is given,
+                # fail here
+                # TODO: Is this condition valid for != '.' only?
+                path_exists = True
+                cmd = ssh_cmd + ["ls", path]
+                try:
+                    out, err = runner.run(cmd, expect_fail=True,
+                                          expect_stderr=True)
+                except CommandError as e:
+                    if "%s: No such file or directory" % path in e.stderr:
+                        path_exists = False
+                    else:
+                        raise  # It's an unexpected failure here
+
+                if path_exists and not force:
+                    raise RuntimeError("Target directory %s already exists." %
+                                       path)
+
+                cmd = ssh_cmd + ["mkdir", "-p", path]
+                try:
+                    runner.run(cmd)
+                except CommandError as e:
+                    lgr.error("Remotely creating target directory failed at "
+                              "%s.\nError: %s" % (path, str(e)))
+                    continue
+
+            # init git repo
+            cmd = ssh_cmd + ["git", "-C", path, "init"]
+            try:
+                runner.run(cmd)
+            except CommandError as e:
+                lgr.error("Remotely initializing git repository failed at %s."
+                          "\nError: %s" % (path, str(e)))
+                continue
+
+            # allow for pushing to checked out branch
+            cmd = ssh_cmd + ["git", "-C", path, "config",
+                             "receive.denyCurrentBranch",
+                             "updateInstead"]
+            try:
+                runner.run(cmd)
+            except CommandError as e:
+                lgr.warning("git config failed at remote location %s.\n"
+                            "Skipped." % path)
+
+            # add the remote
+            cmd = ["git", "remote", "add", target,
+                   target_url.replace("%NAME", REPO_NAME)]
+            runner.run(cmd, cwd=repos[repo].path)
+            cmd = ["git", "remote", "set-url", "--push", target,
+                   target_pushurl.replace("%NAME", REPO_NAME)]
```

Comment:
Maybe it should not set an explicit push URL unless --target-url or --target-push-url were given. It should not hurt, but Git doesn't do it by default, so neither should we.